Refactor and add ability to have 2-5 players

ksk5280 opened the pull request:
Add animations class and domItems module to extract code from game class. Animations are for transitions between sections before the start of the game.

Move meeples when board grows.

Add jQuery and GSAP TweenMax files instead of using CDN's.

Add functionality for choosing between 2-5 players.

closes #33 
closes #35 
closes #39 
closes #42

ksk5280 committed:
Create dom and animations classes to extract code from game class.

ksk5280 committed:
Add animation for game start

ksk5280 committed:
Remove gameShowAnimationCompleted callback

scottfirestone commented:
I'm looking at it in development and the place meeple / skip meeple buttons are missing. Is this a bug or change to functionality?

ksk5280 committed:
Change button displays and refactor game start

ksk5280 commented:
The buttons show now and I also refactored the game start function that was getting really big.

ksk5280 committed:
Add cdns and move meeples when board grows.

ksk5280 committed:
Truncate names that are over 12 characters

ksk5280 committed:
Add player select before game starts

Select dropdown for choosing number of players
Selected number creates like number of player name input fields

ksk5280 committed:
Game builds multiple players

ksk5280 committed:
Game rotates through multiple players

ksk5280 committed:
Tile change border color

ksk5280 committed:
Add colors to player info boxes and dim initially

scottfirestone commented:
Yay more than 2 players! We'll have to fix the audio player but can handle that with another branch.
